qiita_pet/handlers/api_proxy/prep_template.py

```diff
@@ -6,52 +6,358 @@
+            new_investigation_type)
+        prep.investigation_type = investigation_type
+
+    msg = ''
+    status = 'success'
+    if prep_template:
+        fp = check_fp(study_id, prep_template)
+        if isinstance(fp, dict):
+            # Unknown filepath, so return the error message
+            return fp
+        try:
+            with warnings.catch_warnings(record=True) as warns:
+                pt = PrepTemplate(int(prep_id))
+                df = load_template_to_dataframe(fp)
+                pt.extend(df)
+                pt.update(df)
+                remove(fp)
+
+                # join all the warning messages into one. Note that this info
+                # will be ignored if an exception is raised
+                if warns:
+                    msg = '\n'.join(set(str(w.message) for w in warns))
+                    status = 'warning'
+
+        except Exception as e:
+            # Some error occurred while processing the sample template
+            # Show the error to the user so they can fix the template
+            status = 'error'
+            msg = str(e)
+    return {'status': status,
+            'message': msg,
+            'file': prep_template}
+
+
+def prep_template_delete_req(prep_id, user_id):
+    """Delete the prep template
+
+    Parameters
+    ----------
+    prep_id : int
+        The prep template to update
+    user_id : str
+        The current user object id
+    """
+    prep = PrepTemplate(int(prep_id))
+    access_error = check_access(prep.study_id, user_id)
+    if access_error:
+        return access_error
+
+    msg = ''
+    status = 'success'
+    try:
+        PrepTemplate.delete(prep.id)
+    except Exception as e:
+        msg = ("Couldn't remove prep template: %s" % str(e))
+        status = 'error'
+
+    return {'status': status,
+            'message': msg}
+
+
+@execute_as_transaction
+def prep_template_filepaths_get_req(prep_id, user_id):
+    """Returns all filepaths attached to a prep template
+
+    Parameters
+    ----------
+    prep_id : int
+        The current prep template id
+    user_id : int
+        The current user object id
+    """
+    prep = PrepTemplate(int(prep_id))
+    access_error = check_access(prep.study_id, user_id)
+    if access_error:
+        return access_error
+    return {'status': 'success',
+            'message': '',
+            'filepaths': prep.get_filepaths()
             }
-            prep_info[dtype].append(info)
+
+
+def prep_ontology_get_req():
+    """Returns all system and user defined terms for prep template type
+
+    Returns
+    -------
+    dict of list of str
+        {'ENA': [term1, term2, ...],
+         'User': [userterm1, userterm2, ...]}
+    """
+    # Get all the ENA terms for the investigation type
+    ontology = Ontology(convert_to_id('ENA', 'ontology'))
+    # make "Other" last on the list
+    ena_terms = sorted(ontology.terms)
+    ena_terms.remove('Other')
```

Comment:
OK. That means that is always there, right? Cause if not it's gonna fail. So what about always removing and adding in the template?